### PR TITLE
Add custom `:key-fn` for jwt unsign method

### PR DESCRIPTION
**Pull request by just-sultanov**

### CODE REVIEW

**Comment by kipz** [comment]
@just-sultanov I've got a PR open that fixes the build #78 , so you may want to pinch that commit...

Having said that, I'm wondering if anybody is looking after this project at the moment.

**Comment by just-sultanov** [comment]
@kipz Oh, I ran all tests locally on openjdk8. If your changes are accepted, I will pull up from the master branch and update this PR. Thanks

File: src/buddy/sign/jwt.clj [hunk 1]
```diff
@@ -117,11 +117,11 @@
 
 (defn unsign
   ([message pkey] (unsign message pkey {}))
-  ([message pkey {:keys [skip-validation] :or {skip-validation false} :as opts}]
+  ([message pkey {:keys [skip-validation key-fn] :or {skip-validation false key-fn true} :as opts}]
    (try
      (let [claims (-> (jws/unsign message pkey opts)
                       (codecs/bytes->str)
-                      (json/parse-string true))]
+                      (json/parse-string key-fn))]
```
Comment (mjhanninen) [inline src/buddy/sign/jwt.clj]: I'd suggest wrapping `key-fn`, when explicitly given, inside a function that injects an intercepting conversion for the [registered claim names](https://tools.ietf.org/html/rfc7519#section-4.1). These claim names need to be keywords for the claim validation to work properly.

File: test/buddy/sign/jwt_tests.clj [hunk 2]
```diff
@@ -178,3 +179,8 @@
       (is false "unsign should throw")
       (catch clojure.lang.ExceptionInfo e
         (is (= (:cause (ex-data e)) :signature))))))
+
+(deftest jwt-unsign-key-fn-example
+   (let [jwt (jwt/sign {:foo 1 :bar 2 :baz 3} secret {:alg :hs256})
+         actual (jwt/unsign jwt secret {:alg :hs256 :key-fn str/upper-case})]
+     (is (= actual {"FOO" 1 "BAR" 2 "BAZ" 3}))))
```
Comment (mjhanninen) [inline test/buddy/sign/jwt_tests.clj]: I'd suggest the following additional tests:
```clojure
(deftest jwt-unsign-key-fn-claims-checks
  (let [jwt (jwt/sign {:iat 1 :nbf 2 :exp 3} secret {:alg :hs256})]
    (is (thrown-with-msg? clojure.lang.ExceptionInfo #"Token is not yet valid \(2\)"
          (jwt/unsign jwt secret {:alg :hs256
                                  :key-fn str/upper-case
                                  :now 1}))
        "custom key-fn doesn't interfere with nbf claim check")
    (is (thrown-with-msg? clojure.lang.ExceptionInfo #"Token is expired \(3\)"
          (jwt/unsign jwt secret {:alg :hs256
                                  :key-fn str/upper-case
                                  :now 3}))
        "custom key-fn doesn't interfere with exp claim check")))
```

**Comment by niwinz** [comment]
Sorry for the soo late response from my side. I'll review the both PR and merge they in this days ;)

**Comment by kipz** [comment]
Fixed?

**Pull request closed without merging**